reana_client/cli/workflow.py

```diff
@@ -576,10 +570,89 @@ def workflow_run(ctx, file, filenames, name, skip_validation,
                parameter=parameter)
 
 
+@click.command(
+    'delete',
+    help='Delete a workflow run. By default removes all cached'
+         'information of the given workflow and hides it from'
+         'the workflow list.\n'
+         'Workspaces of deleted workflows'
+         'are accessible to retrieve files, to remove the workspace'
+         ' too pass --files flag.\n'
+         'By passing -r/--every-run all workflows with the same'
+         ' will be deleted.\n'
+         'The -a/--all flag will hard delete'
+         ' the workflow data from the database and remove its workspace.')
+@click.option(
+    '-r',
```

Comment:
For dangerous options, we should not make a one-letter shortcut.  Similarly to how `dpkg` uses only long forms for `--force-*` commands.

Since we are in the context of deleting workflows, the option to delete all runs could be simply:

```console
$ reana-client delete -w myanalysis --all-runs
```

I would not use any one-letter shortcut for this dangerous command...